### ISSUE TITLE
SQL Server doesn't correctly serialize TIME bind values in FOR XML and FOR JSON usage

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12181]
When running this:

```java
try (PreparedStatement s = connection.prepareStatement(
    """
    select ? [a]
    for xml raw, elements
    """
)) {
    s.setTime(1, Time.valueOf("10:15:00"));

    try (ResultSet rs = s.executeQuery()) {
        while (rs.next()) {
            System.out.println(rs.getString(1));
        }
    }
}
```

We get:

```xml
<row><a>1970-01-01T10:15:00</a></row>
```

If we explicitly `cast(? as time)`, then the result is correct:

```xml
<row><a>10:15:00</a></row>
```

The JDBC driver seems to send the wrong data type information (?). Perhaps other serialisation formats are affected, too, but definitely `FOR XML` and `FOR JSON` are, which are used in our `MULTISET` emulation

----
See also:
https://github.com/microsoft/mssql-jdbc/issues/1616

**Comment by lukaseder** [comment]
We probably just have to cast all `TIME` bind values to `TIME` all the time.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.1 (https://github.com/jOOQ/jOOQ/issues/12182)
